### ISSUE TITLE
average loss: nan

### ISSUE THREAD

**Issue by earthat** [davisking/dlib#673]
I am doing multiclass classification with dnn network. I have used following configuration

using net_type = loss_multiclass_log<
                                fc<6,
                                relu<fc<45,
                                relu<fc<120,
                                max_pool<2,2,2,2,relu<con<16,5,5,1,1,
                                max_pool<2,2,2,2,relu<con<6,3,3,1,1,
                                input<matrix<float>>

The average loss during training comes out to be Nan in each epoch and not changing. What can be solution or reason behind it.

**Comment by davisking** [comment]
That can happen.  Maybe your learning rate is too high or there is something wrong with your training data.
